aptos-move/block-executor/src/view.rs

```diff
@@ -740,7 +737,12 @@ impl<'a, T: Transaction, S: TStateView<Key = T::Key>, X: Executable> LatestView<
                                         patched_state_value
                                     },
                                     (_, maybe_state_value, _) => {maybe_state_value}
-                                }
+                                };
+                                state.unsync_map.write(state_key.clone(),
```

Comment:
Why move here? We should only write patched values

Reply:
can you explain why this change? was there any test /invariant that required this change?

Reply:
For the parallel case, we are writing the fetched resource value to versioned_map irrespective of whether layout is Some(.) or None. So, I changed the sequential case to do the same. Should I revert back?

Reply:
I see, I had this question for one of your tests in parallel - as seems like we add `None` in readset but the state value does not exist in storage. I think we should fix parallel instead, and keep sequential as is @igor-aptos ?

Reply:
I'll take a look in more depth tomorrow, but just thinking about it - isn't it a very different situation?

In parallel, we use CapturedReads to know which speculative reads were valid or not - and for those we need to capture all reads. Then, we reuse that info - for identity exchange in the end

for sequential execution - without identity exchange - we don't need any read information. so we can store here only if move layout is present - no need to store anything else ?